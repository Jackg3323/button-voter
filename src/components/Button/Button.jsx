import PropTypes from "prop-types";

export default function Button({ changeHandler, txt, bgColor }) {
  return (
    <button
      type="button"
      onClick={changeHandler}
      className={`rounded text-white bg-${bgColor}-500`}
    >
      {txt}
    </button>
  );
}

Button.propTypes = {
  changeHandler: PropTypes.func.isRequired,
  bgColor: PropTypes.string,
  txt: PropTypes.string.isRequired,
};

Button.defaultProps = {
  bgColor: "green",
};
