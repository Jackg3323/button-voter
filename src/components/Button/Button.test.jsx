import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";

it("calls the change handler whenever the button is clicked", () => {
  const mockChangeHandler = jest.fn();

  render(<Button changeHandler={mockChangeHandler} />);

  userEvent.click(screen.getByRole("button"));

  expect(mockChangeHandler).toHaveBeenCalled();
});
