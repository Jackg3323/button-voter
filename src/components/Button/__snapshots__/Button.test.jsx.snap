// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  className="rounded text-white bg-green-500"
  onClick={[Function]}
  type="button"
>
  Click Me!
</button>
`;
